### PR TITLE
Feature/audit logs

### PR DESCRIPTION
**Pull request by TangoBeeAkto**

### CODE REVIEW

File: apps/dashboard/src/main/java/com/akto/action/ApiAuditLogsAction.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    private int startTimestamp;
+    private int endTimestamp;
+
+    private List<ApiAuditLogs> apiAuditLogs;
+    private long totalAuditLogs;
+    public String fetchApiAuditLogsFromDb() {
+        User user = getSUser();
+        if(user == null) return ERROR.toUpperCase();
+
+        RBAC.Role currentRoleForUser = RBACDao.getCurrentRoleForUser(user.getId(), Context.accountId.get());
```
Comment (Ark2307) [inline apps/dashboard/src/main/java/com/akto/action/ApiAuditLogsAction.java:28]: this condition is already in rbac interceptor if this becomes admin actions

File: apps/dashboard/src/main/java/com/akto/interceptor/RoleAccessInterceptor.java [hunk 2]
```diff
@@ -116,12 +131,35 @@ public String intercept(ActionInvocation invocation) throws Exception {
                 ((ActionSupport) invocation.getAction()).addActionError("The role '" + userRole + "' does not have access.");
                 return FORBIDDEN;
             }
+
+            try {
+                if (this.accessType.equalsIgnoreCase(ReadWriteAccess.READ_WRITE.toString())) {
+                    long timestamp = Context.now();
+                    String apiEndpoint = invocation.getProxy().getActionName();
```
Comment (Ark2307) [inline apps/dashboard/src/main/java/com/akto/interceptor/RoleAccessInterceptor.java:138]: add null check

File: apps/dashboard/src/main/java/com/akto/interceptor/RoleAccessInterceptor.java [hunk 3]
```diff
@@ -116,12 +131,35 @@ public String intercept(ActionInvocation invocation) throws Exception {
...
+                    String actionDescription = this.actionDescription == null ? "Error: Description not available" : this.actionDescription;
+                    String userEmail = user.getLogin();
+                    String userAgent = request.getHeader("User-Agent") == null ? "Unknown User-Agent" : request.getHeader("User-Agent");
+                    List<String> userProxyIpAddresses = AuditLogsUtil.getClientIpAddresses(request);
+                    String userIpAddress = userProxyIpAddresses.get(0);
```
Comment (Ark2307) [inline apps/dashboard/src/main/java/com/akto/interceptor/RoleAccessInterceptor.java:143]: add empty arraylist check

File: apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/settings/audit_logs/AuditLogs.jsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        await settingRequests.fetchApiAuditLogsFromDb(0, -1, 1, startTimestamp, endTimestamp).then((resp) => {
+            const apiAuditLogsArr = resp?.apiAuditLogs.map((item) => {
+                const { apiEndpoint, actionDescription, timestamp, userAgent } = item
+                return {
+                    ...item,
+                    apiEndpoint: func.formatEndpoint(apiEndpoint),
+                    actionDescription: <Box width='250px'><p style={{whiteSpace: 'pre-wrap'}}>{actionDescription}</p></Box>,
+                    timestamp: func.prettifyEpoch(timestamp),
+                    rawTimestamp: timestamp,
+                    userAgent: <Box width='fit-content'><p style={{whiteSpace: 'pre-wrap'}}>{userAgent}</p></Box>,
```
Comment (Ark2307) [inline apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/settings/audit_logs/AuditLogs.jsx:90]: use <Text> props

**Review by Ark2307 — APPROVED** [review]

File: libs/dao/src/main/java/com/akto/dao/audit_logs/ApiAuditLogsDao.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        };
+
+        if (!exists) {
+            clients[0].getDatabase(Context.accountId.get()+"").createCollection(getCollName());
+        }
+
+        String[] fieldNames = { ApiAuditLogs.TIMESTAMP };
+        MCollection.createIndexIfAbsent(getDBName(), getCollName(), fieldNames, true);
+
+        fieldNames = new String[]{ApiAuditLogs.USER_EMAIL, ApiAuditLogs.USER_IP_ADDRESS};
```
Comment (avneesh-akto) [inline libs/dao/src/main/java/com/akto/dao/audit_logs/ApiAuditLogsDao.java:31]: add timestamp here too

**Review by avneesh-akto — APPROVED** [review]

**Pull request merged**